src/compiler/crystal/repl/compiler.cr

```diff
@@ -1850,11 +1850,9 @@ class Crystal::Repl::Compiler < Crystal::Visitor
         def: external,
         symbol: @context.c_function(obj_type, external.real_name),
         call_interface: FFI::CallInterface.variadic(
-          abi: FFI::ABI::DEFAULT,
-          args: args_ffi_types,
-          return_type: external.type.ffi_type,
-          fixed_args: external.args.size,
-          total_args: node.args.size,
+          external.type.ffi_type,
+          args_ffi_types,
+          fixed_args: external.args.size
```

Comment:
I wonder if this change made variadic functions stop working?

Reply:
the arguments are the same, so I don't see how can this be. if it worked before, perhaps the issue is in #11475 ?

Reply:
In the old code I see:
- args
- fixed_args
- total_args

In the new code I just see:
- args
- fixed_args

I'm almost sure that's wrong. I can't remember now, but the size of `args` might not be equal to any of `fixed_args` or `total_args`.

Reply:
No. Small types in variable arguments already didn't work with the original FFI implementation. Not sure why you never ran into that (maybe the types are different on macos?).
We need to explicitly promote float and small integer types for variadic function calls, like we're already doing in LLVM codegen. I have this implemented already and will post as a follow-up.

Reply:
The error message from libffi is clear. It's not confused about the numbers of arguments, it just complains about small integer types being used as variable arguments which is simply invalid. Type promotion fixes that.

Reply:
Oh, I see! Yeah, I guess in mac the function is different.